src/libs/actions/PaymentMethods.js

```diff
@@ -209,31 +210,53 @@ function clearDebitCardFormErrorAndSubmit() {
 
 /**
  * Call the API to transfer wallet balance.
+ * @param {Number} transferAmount
  * @param {Object} paymentMethod
  * @param {*} paymentMethod.methodID
  * @param {String} paymentMethod.accountType
  */
-function transferWalletBalance(paymentMethod) {
+function transferWalletBalance(transferAmount, paymentMethod) {
     const paymentMethodIDKey = paymentMethod.accountType === CONST.PAYMENT_METHODS.BANK_ACCOUNT
         ? CONST.PAYMENT_METHOD_ID_KEYS.BANK_ACCOUNT
         : CONST.PAYMENT_METHOD_ID_KEYS.DEBIT_CARD;
     const parameters = {
         [paymentMethodIDKey]: paymentMethod.methodID,
+        shouldReturnOnyxData: true,
     };
-    Onyx.merge(ONYXKEYS.WALLET_TRANSFER, {loading: true});
 
-    DeprecatedAPI.TransferWalletBalance(parameters)
-        .then((response) => {
-            if (response.jsonCode !== 200) {
-                throw new Error(response.message);
-            }
-            Onyx.merge(ONYXKEYS.USER_WALLET, {currentBalance: 0});
-            Onyx.merge(ONYXKEYS.WALLET_TRANSFER, {shouldShowConfirmModal: true, loading: false});
-            Navigation.navigate(ROUTES.SETTINGS_PAYMENTS);
-        }).catch(() => {
-            Growl.error(Localize.translateLocal('transferAmountPage.failedTransfer'));
-            Onyx.merge(ONYXKEYS.WALLET_TRANSFER, {loading: false});
-        });
+    API.write('TransferWalletBalance', parameters, {
+        optimisticData: [
+            {
+                onyxMethod: 'merge',
+                key: ONYXKEYS.WALLET_TRANSFER,
+                value: {
+                    transferAmount,
```

Comment:
Do we still use the transferAmount here as well?

Reply:
Thanks for catching that! Removed.